pkg/dataset/parse.go

```diff
@@ -0,0 +1,141 @@
+package dataset
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/tidwall/gjson"
+)
+
+type metadata struct {
+	GptscriptMetadata bool   `json:"gptscriptMetadata"`
+	File              string `json:"file"`
+	Method            string `json:"method,omitempty"`
+	Splitter          string `json:"splitter,omitempty"`
+}
+
+func ParseDataset(id, workspace string) (Dataset, error) {
+	idFile := workspace + string(os.PathSeparator) + id
+
+	idInfo, err := os.Stat(idFile)
+	if err != nil {
+		return nil, fmt.Errorf("error getting info for dataset %s: %v", id, err)
+	}
+
+	if idInfo.IsDir() {
+		return parseDir(id, workspace)
+	} else if idInfo.Size() > 100*1024*1024 { // 100 MiB
```

Comment:
nit: the compiler probably already does this, but you have the comment explaining what this number is.
```suggestion
	} else if idInfo.Size() > 104857600 { // 100 MiB
```

Reply:
no longer relevant after the reimplementation